elastic/apm-server
# ingest: translate known OpenTelemetry JVM metrics

## Motivation/summary

Copy well-known JVM metrics from their OpenTelemetry
names and associated labels to their counterparts
produced by the Elastic APM Java agent. This enables
users to visualise the metrics in the Elastic APM
app in Kibana.

Metrics are duplicated to avoid surprising users by
silently dropping the OpenTelemetry names. Users can
drop either one of the duplicates with ingest pipeline
customisation.

## Checklist


## How to test these changes

1. Run apm-server
2. Instrument a Java application with https://github.com/open-telemetry/opentelemetry-java-instrumentation, pointed at apm-server
3. Wait for metrics to be collected and sent to APM Server (not sure how long this is)
4. Navigate to the APM app in Kibana, and check the GC and memory usage graphs are populated

See also https://github.com/elastic/apm-server/issues/4919

## Related issues

Closes https://github.com/elastic/apm-server/issues/4919
